skactiveml/base.py

```diff
@@ -1331,6 +1331,188 @@ def _validate_data(
+                "ensure_min_samples": 0,
+                "ensure_min_features": 0,
+                "ensure_2d": False,
+                "force_all_finite": False,
+                "dtype": None,
+            }
+
+        check_missing_label(self.missing_label)
+        self.missing_label_ = self.missing_label
+
+        # Store and check random state.
+        self.random_state_ = check_random_state(self.random_state)
+
+        X = check_array(X, **check_X_dict)
+        y = check_array(y, **check_y_dict)
+        if len(y) > 0:
+            y = column_or_1d(y) if y_ensure_1d else y
+
+        if sample_weight is not None:
+            sample_weight = check_array(sample_weight, **check_y_dict)
+            if not np.array_equal(y.shape, sample_weight.shape):
+                raise ValueError(
+                    f"`y` has the shape {y.shape} and `sample_weight` has the "
+                    f"shape {sample_weight.shape}. Both need to have "
+                    f"identical shapes."
+                )
+
+        return X, y, sample_weight
+
+
+class ProbabilisticRegressor(SkactivemlRegressor):
+    """ProbabilisticRegressor
+
+    Bass class for scikit-activeml probabilistic regressors.
+
+    """
+
+    @abstractmethod
+    def predict_target_distribution(self, X):
+        """Returns the predicted target distribution conditioned on the test
+        samples `X`.
+
+        Parameters
+        ----------
+        X :  array-like, shape (n_samples, n_features)
+            Input samples.
+
+        Returns
+        -------
+        dist : scipy.stats._distn_infrastructure.rv_frozen
+            The distribution of the targets at the test samples.
+
+        """
+        raise NotImplementedError
+
+    def predict(self, X, return_std=False, return_entropy=False):
+        """Returns the mean, std (optional) and differential entropy (optional)
+        of the predicted target distribution conditioned on the test samples
+        `X`.
+
+        Parameters
+        ----------
+        X :  array-like, shape (n_samples, n_features)
+            Input samples.
+        return_std : bool, optional (default=False)
+            Whether to return the standard deviation.
+        return_entropy : bool, optional (default=False)
+            Whether to return the differential entropy.
+
+        Returns
+        -------
+        mu : numpy.ndarray, shape (n_samples)
+            Predicted mean conditioned on `X`.
+        std : numpy.ndarray, shape (n_samples), optional
+            Predicted standard deviation conditioned on `X`.
+        entropy : numpy..ndarray, optional
+            Predicted differential entropy conditioned on `X`.
+        """
+        rv = self.predict_target_distribution(X)
+        result = (rv.mean(),)
+        if return_std:
+            result += (rv.std(),)
+        if return_entropy:
+            result += (rv.entropy(),)
+        if len(result) == 1:
+            result = result[0]
+        return result
```

Comment:
I do not think that this is a good design pattern: A function that sometimes return 1, 2 or 3 values.

I would suggest always returning `std` and `entropy` . Then the other dev calling this function can use `_` to ignore the params they do not want.

Reply:
see also https://stackoverflow.com/questions/44044932/how-can-a-function-optionally-return-one-or-more-values